streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredKeyValueStore.java

```diff
@@ -171,42 +169,24 @@ private void registerMetrics() {
     @Deprecated
     private void initStoreSerde(final ProcessorContext context) {
         final String storeName = name();
-        final String changelogTopic = ProcessorContextUtils.changelogFor(context, storeName);
-        final String prefix = getPrefix(context.appConfigs(), context.applicationId());
+        final String changelogTopic = ProcessorContextUtils.changelogFor(context, storeName, Boolean.FALSE);
```

Comment:
Only three MeteredXYZStore classes' `initStoreSerde` have the third param to false, but it seems that if the returned value is `null` we would still fallback to `ProcessorStateManager.storeChangelogTopic(prefix, storeName, taskId.topologyName())` to create the new topic, so it should be true as well??

BTW if my other comment is valid, then we can remove this param anyways.

Reply:
What I understood going through the code is that in MeteredXYZStore classes first, we check if the returned value is null then we create a new topic, and in classes, like WindowCacheStore and InMemoryTimeOrderedKeyValueBuffer we always need to create a new topic.
Hence I introduced the third param which when false goes through the whole procedure of checking null and creating a new topic, and in case the param is set to True it just creates a new topic.

Reply:
Hi, after going through the code I realized that the classes like CachingWindowStore and InMemoryTimeOrderedKeyValueBuffer always need to create a new topic whereas the classes MeteredXYZStore have a null checking code flow.
As there are two types of flow. So, I thought to have a bool variable explicitly to have both flows in the same function.

Reply:
Yes, but I'm wondering if the two types of flow are really necessary. It seems to me that the topic should have always been created at the `init` phase already, and hence it seems the null check is not necessary.

@cadonna Would need your quick thoughts about this.